common/src/main/java/us/ilite/common/lib/control/PIDController.java

```diff
@@ -0,0 +1,331 @@
+    private double m_totalError = 0.0; // the sum of the errors for use in the
+                                       // integral calc
+    private double m_setpoint = 0.0;
+    private double m_error = 100.0;
+    private double m_result = 0.0;
+    private double m_last_input = Double.NaN;
+    private double m_deadband = 0.0; // If the absolute error is less than
+                                     // deadband
+                                     // then treat error for the proportional
+                                     // term as 0
+
+    public PIDController() {
+    }
+
+    /**
+     * Allocate a PID object with the given constants for P, I, D
+     *
+     * @param Kp
+     *            the proportional coefficient
+     * @param Ki
+     *            the integral coefficient
+     * @param Kd
+     *            the derivative coefficient
+     */
+    public PIDController(double Kp, double Ki, double Kd) {
+        m_P = Kp;
+        m_I = Ki;
+        m_D = Kd;
+        m_F = 0;
+    }
+
+    /**
+     * Allocate a PID object with the given constants for P, I, D
+     *
+     * @param Kp
+     *            the proportional coefficient
+     * @param Ki
+     *            the integral coefficient
+     * @param Kd
+     *            the derivative coefficient
+     * @param Kf
+     *            the feed forward gain coefficient
+     */
+    public PIDController(double Kp, double Ki, double Kd, double Kf) {
+        m_P = Kp;
+        m_I = Ki;
+        m_D = Kd;
+        m_F = Kf;
+    }
+
+    /**
+     * Read the input, calculate the output accordingly, and write to the output. This should be called at a constant
+     * rate by the user (ex. in a timed thread)
+     *
+     * @param input
+     *            the input
+     * @param dt
+     *            time passed since previous call to calculate
+     */
+    public double calculate(double input, double dt) {
+        if (dt < 1E-6)
+            dt = 1E-6;
+        m_last_input = input;
+        m_error = m_setpoint - input;
+        
+        if (m_continuous) {
+            if (Math.abs(m_error) > (m_maximumInput - m_minimumInput) / 2) {
+                if (m_error > 0) {
+                    m_error = m_error - m_maximumInput + m_minimumInput;
+                } else {
+                    m_error = m_error + m_maximumInput - m_minimumInput;
+                }
+            }
+        }
+
+        if ((m_error * m_P < m_maximumOutput) && (m_error * m_P > m_minimumOutput)) {
+            m_totalError += m_error * dt;
+        } else {
+            m_totalError = 0;
+        }
+
+        // Don't blow away m_error so as to not break derivative
+        double proportionalError = Math.abs(m_error) < m_deadband ? 0 : m_error;
+
+        m_result = (m_P * proportionalError + m_I * m_totalError + m_D * (m_error - m_prevError) / dt
+                + m_F * m_setpoint);
+        m_prevError = m_error;
+
+        if (m_result > m_maximumOutput) {
```

Comment:
I think we have a clamp() method that does this functionality.  That would get rid of needing to track 2 variables in this class, as well as replicating that function here.

Reply:
According to @StephenWelch the method is: 

`com.team254.lib.util.Util.limit`